clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/KafkaStoreIngestionService.java

```diff
@@ -1325,6 +1325,11 @@ public MetadataResponse getMetadata(String storeName) {
       response.setLatestSuperSetValueSchemaId(latestSuperSetValueSchemaId);
       response.setRoutingInfo(routingInfo);
       response.setHelixGroupInfo(helixGroupInfo);
+      if (store.getBatchGetLimit() > 0) {
```

Comment:
IDK the background here so maybe there is a good reason... but why is this function part of the `KafkaStoreIngestionService`? It does not seem to make sense to have it here 🤔 ...

Reply:
I guess it's part of the `MetadataRetriever` interface, so that's why... Some of the functions in that interface (`getConsumptionSnapshots` and `getTopicPartitionIngestionContext`) are related to ingestion, so we probably need to keep that interface in any case... while the other 3 functions (`getStoreVersionCompressionDictionary`, `getMetadata` and `getCurrentVersionResponse`) are accessing objects which are passed into the `KSIS` constructor by the `VeniceServer` (or other code paths...) so it might make sense to pass those objects into a different container and have the remaining three functions be decoupled from `KSIS`, but IDK if it's worth it... Looks like the `schemaRepo`, `customizedViewRepository` and `helixInstanceConfigRepository` are only used by `getMetadata` so we could remove those completely from `KSIS` if we decoupled... It seems a tiny bit cleaner, but IDK if it's worth the effort. WDYT?

Reply:
Yeah I took a look and it's not super clean since we still need `KSIS` to implement `MetadataRetriever` to provide the two consumption related metadata as you mentioned. I don't mind doing some cleanup if we are ok with another service class that solely acts as a container to share some reference for the repositories to provide metadata lookup.

Reply:
Well... I guess it'd be nice to move the latter 3 functions to a different class, since `KSIS` is already fairly big, at 1400 LoC, so it's gravitating towards God class territory. So splitting the `MetadataRetriever` interface into two, e.g. perhaps `IngestionMetadataRetriever` and `ReadMetadataRetriever` or something like that, and being able to remove some of the objects passed into the `KSIS` constructor (which is also quite large at this point) might help tidy things up a bit. Personally, I think it's ok to spend a bit more time to leave the codebase in a cleaner state than we found it, but I leave it up to you to make the call.

Reply:
I agree, will make the refactor and changes.